src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8513,44 +8519,198 @@ protected MethodGroupResolution ResolveExtension(
+                    lookupResult, left.Type, memberName, arity,
+                    basesBeingResolved: null, lookupOptions, originalBinder: binder, ref useSiteInfo);
+
+                CompoundUseSiteInfo<AssemblySymbol> classicExtensionUseSiteInfo = binder.GetNewCompoundUseSiteInfo(diagnostics);
+                binder.LookupExtensionMethods(classicExtensionLookupResult, scope, memberName, arity, ref classicExtensionUseSiteInfo);
                 diagnostics.Add(expression, classicExtensionUseSiteInfo);
 
                 lookupResult.MergeEqual(classicExtensionLookupResult);
 
                 if (!lookupResult.IsMultiViable)
                 {
-                    diagnostics.Free();
-                    continue;
+                    return false;
                 }
 
-                var members = ArrayBuilder<Symbol>.GetInstance();
-                Symbol? symbol = this.GetSymbolOrMethodOrPropertyGroupStrict(lookupResult, expression, memberName, arity, members, diagnostics, qualifierOpt: null);
-                if (symbol is not null)
+                // 2. resolve methods
+                MethodGroupResolution methodResult = resolveMethods(expression, left, typeArgumentsWithAnnotations, returnType, returnRefKind,
+                    lookupResult, analyzedArguments, ref actualMethodArguments, options, in callingConvention, binder, diagnostics);
+
+                // 3. resolve properties
+                OverloadResolutionResult<PropertySymbol>? propertyResult = resolveProperties(left, lookupResult, binder, ref actualReceiverArguments, ref useSiteInfo);
+
+                // 4. determine member kind
+                if (!methodResult.HasAnyApplicableMethod && propertyResult?.HasAnyApplicableMember != true)
                 {
-                    firstResult.Free();
-                    actualArguments?.Free();
-                    lookupResult.Free();
-                    classicExtensionLookupResult.Free();
-                    members.Free();
-                    return new MethodGroupResolution(symbol, LookupResultKind.Viable, diagnostics.ToReadOnlyAndFree());
+                    // Found nothing applicable. Store aside the first non-applicable result and continue searching for an applicable result.
+                    if (firstResult.IsEmpty)
+                    {
+                        if (propertyResult != null)
+                        {
+                            methodResult.Free(keepArguments: true);
+                            propertyResult.Free();
+                            firstResult = makeErrorResult(left.Type, memberName, arity, lookupResult, expression, diagnostics);
+                        }
+                        else
+                        {
+                            firstResult = methodResult;
+                        }
+                    }
+
+                    return false;
+                }
+
+                if (methodResult.HasAnyApplicableMethod)
```

Comment:
>HasAnyApplicableMethod [](http://example.com/codeflow?start=33&length=22)

Is there a reason to use this strange helper? it looks like it can return `true` without ```OverloadResolutionResult``` present.